daemon/cluster/listen_addr.go

```diff
@@ -87,6 +89,38 @@ func (c *Cluster) resolveAdvertiseAddr(advertiseAddr, listenAddrPort string) (st
 	return systemAddr.String(), listenAddrPort, nil
 }
 
+// validateDefaultAddrPool validates default address pool
+func validateDefaultAddrPool(defaultAddrPool []string, size uint32) error {
+	if defaultAddrPool == nil {
+		// defaultAddrPool is not defined
+		return nil
+	}
+	//if size is not set, then we use default value 24
+	if size == 0 {
+		size = 24
+	}
+	if size > 32 {
+		return fmt.Errorf("subnet size is out of range: %d", size)
+	}
+	for i := range defaultAddrPool {
+		//trim leading and trailing white spaces
+		defaultAddrPool[i] = strings.TrimSpace(defaultAddrPool[i])
+		result := strings.Split(defaultAddrPool[i], "/")
```

Comment:
I thought we went back and forth converting string to IPNet and that was one of the code refactoring part.  We had validation code like that in CLI as we were get input as IPNet.   But now here we received as string so dont want to go back and convert to IPNet again just for validation. Pls let me know you still want me to move the logic. I can change it. 

WRT error info, I will update.

Reply:
My concern was that the "custom" validation we're doing is only checking; if there's a single slash in the string, and if the slash is followed by an integer that's smaller than `size`.

But it ignores any other formatting errors (e.g. `hello!@#@#@%#@%world/24` [is considered valid](https://play.golang.org/p/k9_h9VWOd-_C)).

Perhaps that's ok, because it will fail further down the line, but I'm not sure

Reply:
I think I will change validation back to old way ..